abdulfatai360/banka
# #165270815 Enable email notification for transaction endpoint

#### What does this PR do?
Enable real-time email notification when a debit or credit transaction occurs on a user's bank account

#### Description of Task to be completed?
- Have user receive instant email alert when a transaction occurred on his/her account
- Make the email look and feel user friendly

#### How should this be manually tested?
On Terminal:
- Run `git clone https://github.com/abdulfatai360/banka.git` to clone this repo onto your local machine
- Navigate to the repo directory by running `cd banka/`
- Ensure your working directory is clean by running  `git status`  command
- Get a copy of the PR by typing  `git fetch origin pull/37/head:ft-transactional-email-notification-165270815`
- Now that you have a copy of the branch, switch to it using  `git checkout ft-transactional-email-notification-165270815`
- Your directory will now be an exact copy of the PR
- Run `npm install` to install all project's dependencies onto your local machine
- Type `npm run build` to run the build script and populate the `dist/` directory containing transpiled code
- Run `npm start` command to start the development server

On Postman:
Test these endpoints:
- `http://localhost:3000/ap1/v1/transactions/<account-number>/debit` - `HTTP POST`
- `http://localhost:3000/ap1/v1/transactions/<account-number>/credit` - `HTTP POST`

You should receive email alerts if you registered with a working email address.

#### Any background context you want to provide?
Before you test the endpoint on Postman, do the following:

Register a user account: `http://localhost:3000/ap1/v1/auth/signup` - `HTTP POST`
Open a new bank account for the user: `http://localhost:3000/ap1/v1/accounts` - `HTTP POST`
Register a cashier account: `http://localhost:3000/ap1/v1/auth/signup` - `HTTP POST` by having a "type" field of value "client"

#### What are the relevant pivotal tracker stories?
#165270815
